src/main/java/org/datadog/jenkins/plugins/datadog/DatadogPluginManagement.java

```diff
@@ -0,0 +1,144 @@
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Extension
+public class DatadogPluginManagement extends ManagementLink {
+
+    private static final Logger LOGGER = Logger.getLogger(DatadogPluginManagement.class.getName());
+
+    private final List<FlareContributor> contributors;
+
+    public DatadogPluginManagement() {
+        contributors = new ArrayList<>(ExtensionList.lookup(FlareContributor.class));
+        contributors.sort(Comparator.comparingInt(FlareContributor::order));
+    }
+
+    public List<FlareContributor> getContributors() {
+        return contributors;
+    }
+
+    @CheckForNull
+    @Override
+    public String getIconFileName() {
+        return "/plugin/datadog/icons/dd_icon_rgb.svg";
+    }
+
+    @CheckForNull
+    @Override
+    public String getDisplayName() {
+        return "Datadog";
+    }
+
+    @CheckForNull
+    @Override
+    public String getUrlName() {
+        return "datadog";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Datadog Plugin Troubleshooting";
+    }
+
+    @NonNull
+    @Override
+    public Category getCategory() {
+        return Category.TROUBLESHOOTING;
+    }
+
+    @NonNull
+    @Override
+    public Permission getRequiredPermission() {
+        return Jenkins.MANAGE;
+    }
+
+    @RequirePOST
+    public void doDownloadDiagnosticFlare(StaplerRequest request, StaplerResponse response) throws Exception {
+        if (!Jenkins.get().hasPermission(Jenkins.MANAGE)) {
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        try {
+            LocalDateTime now = LocalDateTime.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss");
+            String formattedTimestamp = now.format(formatter);
+
+            response.setContentType("application/octet-stream");
+            response.setHeader("Content-Disposition", String.format("attachment; filename=dd-jenkins-plugin-flare-%s.zip", formattedTimestamp));
+
+            try (OutputStream out = response.getOutputStream()) {
+                List<FlareContributor> selectedContributors = getSelectedContributors(request);
+                writeDiagnosticFlare(selectedContributors, out);
+            }
+
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOGGER.severe("Interrupted while waiting for JFR recording to finish");
```

Comment:
```suggestion
            LOGGER.severe("Interrupted while generating Datadog plugin flare");
```
Would it be like this?